synapse/handlers/e2e_keys.py

```diff
@@ -977,12 +985,66 @@ def _get_e2e_cross_signing_verify_key(self, user_id, key_type, from_user_id=None
         Raises:
             NotFoundError: if the key is not found
         """
+        user = UserID.from_string(user_id)
         key = yield self.store.get_e2e_cross_signing_key(
             user_id, key_type, from_user_id
         )
+
+        if key is None and self.is_mine(user):
+            # Attempt to fetch the missing key from the remote user's server
+            try:
+                remote_result = yield self.federation.query_client_keys(
+                    user.domain, {"device_keys": {user_id: []}}, timeout=10 * 1000
```

Comment:
given that we know the id of the key we're after, can we not just query for that?